src/aws-cpp-sdk-core/source/internal/AWSHttpResourceClient.cpp

```diff
@@ -207,6 +207,30 @@ namespace Aws
             AWS_LOGSTREAM_TRACE(m_logtag.c_str(), "IMDSv1 had been disabled at the SDK build time");
 #endif
         }
+        
+        EC2MetadataClient::EC2MetadataClient(const Aws::Client::ClientConfiguration::CredentialProviderConfiguration& credentialConfig,
+            const char *endpoint) :
+            AWSHttpResourceClient([&credentialConfig]() {
+                Aws::Client::ClientConfiguration clientConfig;
+                clientConfig.credentialProviderConfig = credentialConfig;
+                clientConfig.connectTimeoutMs = credentialConfig.imdsConfig.metadataServiceTimeout * 1000;
+                clientConfig.requestTimeoutMs = credentialConfig.imdsConfig.metadataServiceTimeout * 1000;
+                clientConfig.retryStrategy = Aws::MakeShared<DefaultRetryStrategy>(RESOURCE_CLIENT_CONFIGURATION_ALLOCATION_TAG, credentialConfig.imdsConfig.metadataServiceNumAttempts - 1, 1000);
+                clientConfig.maxConnections = 2;
+                clientConfig.scheme = Scheme::HTTP;
+                return clientConfig;
+            }(), EC2_METADATA_CLIENT_LOG_TAG),
+            m_endpoint(endpoint),
+            m_disableIMDS(false),
+            m_tokenRequired(true),
+            m_disableIMDSV1(false)
+        {
+#if defined(DISABLE_IMDSV1)
+            AWS_UNREFERENCED_PARAM(m_disableIMDSV1);
```

Comment:
why do we have a `AWS_UNREFERENCED_PARAM` on a member variable that is referenced in the constructor above?